packages/status-js/src/ethereum-client/ethereum-client.ts

```diff
@@ -0,0 +1,42 @@
+import { Point } from 'ethereum-cryptography/secp256k1'
+import { ethers } from 'ethers'
+
+export class EthereumClient {
+  private provider: ethers.JsonRpcApiProvider
+
+  constructor(url: string) {
+    this.provider = new ethers.JsonRpcProvider(url)
+  }
+
+  async resolveChatKey(ensName: string): Promise<string | undefined> {
+    try {
+      const resolver = await this.provider.getResolver(ensName)
```

Comment:
🤦‍♂️ thought this wouldn't apply to `stateofus.eth` subdomains, but it turned out I had a typo in my example record.

No need to adjust. Those records would be resolved with a public resolver (`0x5FfC014343cd971B7eb70732021E26C35B744cc4`)

Reply:
Which is an old resolver by the way,

> A new instance of the public resolver has been deployed, and can be found at 0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41. This instance references the new ENS registry, and has an additional ‘multicall’ feature implemented, which permits users to set multiple records in a single operation.
Since the public resolver looks up names in the ENS registry to determine who is permitted to configure records for them, while names - migrated or otherwise - pointed at an old instance of the public resolver will continue to function, they will need to migrate to the new public resolver in order to make changes. The ENS dapp at app.ens.domains facilitates this process using the new multicall function, making it possible to migrate over all records for a name in two transactions: the first one copying all records from the old resolver to the new one, and the second one updating the registry to point to the new resolver contract.
> 
> – https://docs.ens.domains/ens-migration-february-2020/technical-description#public-resolver

@cammellos @richard-ramos by chance, have you considered migrating?

Reply:
cc: @3esmit